src/clp_ffi_py/api_decoration.hpp

```diff
@@ -0,0 +1,11 @@
+#ifndef CLP_FFI_PY_API_DECORATION_HPP
+#define CLP_FFI_PY_API_DECORATION_HPP
+
+/**
+ * `CLP_FFI_PY_METHOD` should be added at the beginning of a function's declaration/implementation
+ * to decorate any APIs that are directly invoked by Python's interpreter. The macro expands to
+ * `extern "C"` to ensure C linkage.
+ */
+#define CLP_FFI_PY_METHOD extern "C"
```

Comment:
Do we need to add this to any existing methods?

Reply:
No. For the current methods, we used `extern "C"` to wrap the entire code block that contains C-linkage methods. As we discussed before we should apply `extern "C"` method by method. This macro is mirroring what we added in Go-ffi (even though in Go ffi the usage is a little more complicated)